### PR TITLE
Adds the possibility to register plugins to capture embed links…

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
Adds the possibility to register plugins that will render a specific embed view for certain links. The plugins will register a match function to check the embed links against, and a component to render if the match function returns true. Lastly, it is possible as well to indicate if the embed view should be collapsable or not.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17395

#### Related Pull Requests
- [Has server changes](https://github.com/mattermost/mattermost-server/pull/11723)

### CODE REVIEW

File: plugins/registry.js [hunk 1]
```diff
@@ -167,6 +167,31 @@ export default class PluginRegistry {
         return id;
     }
 
+    // Register a component to render a custom embed preview for post links.
+    // Accepts the following:
+    // - match - A function that receives the embed link and returns a
```
Comment (lieut-data) [inline plugins/registry.js]: It looks like it actually receives an embed object, with various other data in addition to the link. Can you take a stab at documenting what a plugin author can expect here?

**Comment by lindalumitchell** [comment]
Is this related to a specific ticket / can you help create one? Is this an improvement for v5.16?

File: plugins/registry.js (hunk 1 above)
Comment (mgdelacroix) [reply plugins/registry.js]: Sure thing! I just sent a commit adding the embed object keys and a simple description not to tie the documentation it too much to the backend specifics. Let me know if this is close to what you were thinking or if another format or more extended contents are preferred :)

**Comment by mgdelacroix** [comment]
> Is this related to a specific ticket / can you help create one? Is this an improvement for v5.16?

Created the ticket and set the milestone as per our conversation

**Comment by lindalumitchell** [comment]
QA will review after v5.14 release is out.

**Pull request merged**